Tests/Api/ClientTest.php

```diff
@@ -0,0 +1,312 @@
+<?php
+
+namespace Test\Werkspot\BingAdsApiBundle\Api;
+
+use BingAds\Bulk\ReportTimePeriod;
+use BingAds\Proxy\ClientProxy;
+use Mockery;
+use Symfony\Component\Filesystem\Filesystem;
+use Werkspot\BingAdsApiBundle\Api\Client;
+use Werkspot\BingAdsApiBundle\Api\Exceptions;
+use Werkspot\BingAdsApiBundle\Guzzle\RequestNewAccessToken;
+use Werkspot\BingAdsApiBundle\Api\Helper;
+
```

Comment:
There are  public methods in the Client, like getCacheDir, clearCache that I think should be private, also is setApiDetails called? and it it's manditory, why shouldn't this be passed in the constructor? Now if you don't call this method the get() will fail